# Imports

In [1]:
import pandas as pd
import yaml
import numpy as np
import plotly.express as px
import plotly.graph_objects as go

pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)

# Get Data

In [2]:
config = yaml.safe_load(open('../../configuration/path.yaml'))

In [3]:
df = pd.read_parquet(config['data']['bronze']['parquet'][2022])

# EDA

In [4]:
df

gameid datacompleteness                                          url  league  year   split  playoffs                 date  game  patch  participantid  side position playername                                   playerid                  teamname                                   teamid  champion     ban1     ban2    ban3     ban4   ban5  gamelength  result  kills  deaths  assists  teamkills  teamdeaths  doublekills  triplekills  quadrakills  pentakills  firstblood  firstbloodkill  firstbloodassist  firstbloodvictim  team kpm    ckpm  firstdragon  dragons  opp_dragons  elementaldrakes  opp_elementaldrakes  infernals  mountains  clouds  oceans  chemtechs  hextechs  dragons (type unknown)  elders  opp_elders  firstherald  heralds  opp_heralds  firstbaron  barons  opp_barons  firsttower  towers  opp_towers  firstmidtower  firsttothreetowers  turretplates  opp_turretplates  inhibitors  opp_inhibitors  damagetochampions        dpm  damageshare  damagetakenperminute  \
0       ESPORTSTMNT01_2690210         complete                                         None  LCK CL  2022  Spring         0  2022-01-10 07:44:08     1  12.01              1  Blue      top     Soboro  oe:player:38e0af7278d6769d0c81d7c4b47ac1e  Fredit BRION Challengers  oe:team:68911b3329146587617ab2973106e23  Renekton    Karma  Caitlyn  Syndra   Thresh   Lulu        1713       0      2       3        2          9          19          0.0          0.0          0.0         0.0         0.0             0.0               0.0               0.0    0.3152  0.9807          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     0.0         0.0         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         0.0             0.0            15768.0   552.2942     0.278784             1072.3993   
1       ESPORTSTMNT01_2690210         complete                                         None  LCK CL  2022  Spring         0  2022-01-10 07:44:08     1  12.01              2  Blue      jng     Raptor  oe:player:637ed20b1e41be1c51bd1a4cb211357  Fredit BRION Challengers  oe:team:68911b3329146587617ab2973106e23  Xin Zhao    Karma  Caitlyn  Syndra   Thresh   Lulu        1713       0      2       5        6          9          19          0.0          0.0          0.0         0.0         1.0             0.0               1.0               0.0    0.3152  0.9807          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     0.0         0.0         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         0.0             1.0            11765.0   412.0841     0.208009              944.2732   
2       ESPORTSTMNT01_2690210         complete                                         None  LCK CL  2022  Spring         0  2022-01-10 07:44:08     1  12.01              3  Blue      mid     Feisty  oe:player:d1ae0e2f9f3ac1e0e0cdcb86504ca77  Fredit BRION Challengers  oe:team:68911b3329146587617ab2973106e23   LeBlanc    Karma  Caitlyn  Syndra   Thresh   Lulu        1713       0      2       2        3          9          19          0.0          0.0          0.0         0.0         0.0             0.0               0.0               0.0    0.3152  0.9807          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     0.0         0.0         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         0.0             0.0            14258.0   499.4046     0.252086              581.6462   
3       ESPORTSTMNT01_26902

In [5]:
df.shape

(149400, 123)

This dataset has 149.400 rows and 123 columns.

In [6]:
df.dtypes

gameid                       object
datacompleteness             object
url                          object
league                       object
year                          int64
split                        object
playoffs                      int64
date                         object
game                          int64
patch                       float64
participantid                 int64
side                         object
position                     object
playername                   object
playerid                     object
teamname                     object
teamid                       object
champion                     object
ban1                         object
ban2                         object
ban3                         object
ban4                         object
ban5                         object
gamelength                    int64
result                        int64
kills                         int64
deaths                        int64
assists                     

Now that we are aware of the data types associated with each column in this dataset, let us proceed with a thorough Exploratory Data Analysis (EDA) to refine the dataset. Each column and its associated properties shall be examined in detail.

## Columns 

### gameid

In [31]:
list(df.columns)

['gameid',
 'datacompleteness',
 'url',
 'league',
 'year',
 'split',
 'playoffs',
 'date',
 'game',
 'patch',
 'participantid',
 'side',
 'position',
 'playername',
 'playerid',
 'teamname',
 'teamid',
 'champion',
 'ban1',
 'ban2',
 'ban3',
 'ban4',
 'ban5',
 'gamelength',
 'result',
 'kills',
 'deaths',
 'assists',
 'teamkills',
 'teamdeaths',
 'doublekills',
 'triplekills',
 'quadrakills',
 'pentakills',
 'firstblood',
 'firstbloodkill',
 'firstbloodassist',
 'firstbloodvictim',
 'team kpm',
 'ckpm',
 'firstdragon',
 'dragons',
 'opp_dragons',
 'elementaldrakes',
 'opp_elementaldrakes',
 'infernals',
 'mountains',
 'clouds',
 'oceans',
 'chemtechs',
 'hextechs',
 'dragons (type unknown)',
 'elders',
 'opp_elders',
 'firstherald',
 'heralds',
 'opp_heralds',
 'firstbaron',
 'barons',
 'opp_barons',
 'firsttower',
 'towers',
 'opp_towers',
 'firstmidtower',
 'firsttothreetowers',
 'turretplates',
 'opp_turretplates',
 'inhibitors',
 'opp_inhibitors',
 'damagetochampions',
 'dpm',
 'd

In [7]:
list(df['gameid'].unique())

['ESPORTSTMNT01_2690210',
 'ESPORTSTMNT01_2690219',
 '8401-8401_game_1',
 'ESPORTSTMNT01_2690227',
 '8401-8401_game_2',
 'ESPORTSTMNT01_2690255',
 '8402-8402_game_1',
 'ESPORTSTMNT01_2690264',
 '8402-8402_game_2',
 '8402-8402_game_3',
 'ESPORTSTMNT01_2690302',
 'ESPORTSTMNT01_2700416',
 'ESPORTSTMNT01_2700432',
 'ESPORTSTMNT01_2690328',
 'ESPORTSTMNT01_2690351',
 'ESPORTSTMNT01_2700456',
 'ESPORTSTMNT01_2690370',
 'ESPORTSTMNT01_2700475',
 'ESPORTSTMNT01_2690390',
 'ESPORTSTMNT01_2700491',
 'NA1_4170726256',
 'NA1_4170915669',
 'NA1_4170925713',
 'NA1_4170707765',
 'NA1_4170786291',
 'NA1_4170905921',
 'NA1_4170975598',
 'NA1_4171085502',
 'NA1_4171175222',
 'NA1_4170696128',
 'NA1_4171055557',
 'NA1_4170815921',
 'NA1_4170796573',
 'NA1_4170796592',
 'NA1_4170815896',
 'NA1_4171105636',
 'NA1_4170926184',
 'NA1_4170806914',
 'NA1_4171095985',
 'NA1_4170806921',
 'NA1_4170926227',
 'NA1_4170557244',
 'NA1_4171086065',
 'NA1_4170946197',
 'NA1_4170936357',
 'NA1_4171066428',
 'NA1_41709

There are predominantly 3 ID types and they start with or contains:

1. NA1
2. ESPORTSTMNT
3. 84xx_gamexx

Let us investigate those types and try to identify a pattern.

##### NA1

In [8]:
df[df['gameid'].str.contains('NA1')]['datacompleteness'].unique()

array(['complete'], dtype=object)

In [9]:
df[df['gameid'].str.contains('NA1')]['league'].unique()

array(['Proving Grounds Circuit', 'UPL'], dtype=object)

##### ESPORTSTMNT

In [10]:
df[df['gameid'].str.contains('ESPORTSTMNT')]['datacompleteness'].unique()

array(['complete', 'ignore'], dtype=object)

In [11]:
df[df['gameid'].str.contains('ESPORTSTMNT')]['league'].unique()

array(['LCK CL', 'NLC', 'SL', 'UL', 'PRM', 'LCK', 'LFL',
       'Proving Grounds Circuit', 'LEC', 'LCS', 'LFL2', 'GLL', 'HM',
       'ESLOL', 'EBL', 'LPLOL', 'PGN', 'LCSA', 'LVP DDH', 'TAL', 'TCL',
       'CBLOL', 'LCO', 'LHE', 'GL', 'EL', 'CBLOLA', 'LMF', 'VL', 'LLA',
       'HC', 'LJL', 'PCS', 'VCS', 'LCL', 'NEXO', 'EM', 'LAS', 'MSI',
       'LJLA', 'CT', 'WCS', 'CDF', 'IC'], dtype=object)

##### _game

In [12]:
df[df['gameid'].str.contains('_game')]['datacompleteness'].unique()

array(['partial'], dtype=object)

In [13]:
df[df['gameid'].str.contains('_game')]['league'].unique()

array(['LPL', 'LDL', 'WCS', 'DC'], dtype=object)

It appears that the IDs containing "NA" originate from completed games, "ESPORTSTMNT" from either completed or ignored games, and those containing "game" in their name denote partial matches. Since it is an ID and it won't be done any math with it, it is correctly identified as object into the dataframe.

##### Plotting ID types distribuition

In [14]:
NA1_shape = df[df['gameid'].str.contains('NA1')].shape[0]
ESPORTSTMNT_shape = df[df['gameid'].str.contains('ESPORTSTMNT')].shape[0]
game_shape = df[df['gameid'].str.contains('_game')].shape[0]

data = {'NA': NA1_shape,
        'ESPORTSTMNT': ESPORTSTMNT_shape,
        'GAME': game_shape}

df_plot = pd.DataFrame.from_dict(data, orient = 'index', columns = ['Value'])
df_plot.reset_index(inplace=True)
df_plot.rename(columns = {'index': 'Label'}, inplace=True)

fig = px.bar(df_plot, x = 'Label', y = 'Value', template = 'plotly_dark', text_auto = '.3s')

fig.update_xaxes(categoryorder = 'total descending')

fig.update_layout(
    title = {
        'text': "Quantity of ID Types"},
    xaxis_title = 'ID Types',
    yaxis_title = 'Quantity',
    width = 1200,
    height = 500,
    font_family="Helvetica",
    font_color="White",
    font = dict(size = 18),
    title_font_family="Helvetica",
    title_font_color= "White",
    bargap = 0.4,
    uniformtext_minsize = 8)

fig.update_traces(textposition = 'outside', textfont_size = 16, marker_color = '#C70039')

fig.show()

ValueError: Mime type rendering requires nbformat>=4.2.0 but it is not installed

### datacompleteness

In [16]:
df['datacompleteness'].unique()

array(['complete', 'partial', 'ignore'], dtype=object)

There are tree types of datacompleteness, let us investigate one by one here:

In [17]:
df[df['datacompleteness'] == 'complete']

gameid datacompleteness   url  league  year    split  playoffs                 date  game  patch  participantid  side position playername                                   playerid                    teamname                                   teamid  champion      ban1     ban2    ban3    ban4    ban5  gamelength  result  kills  deaths  assists  teamkills  teamdeaths  doublekills  triplekills  quadrakills  pentakills  firstblood  firstbloodkill  firstbloodassist  firstbloodvictim  team kpm    ckpm  firstdragon  dragons  opp_dragons  elementaldrakes  opp_elementaldrakes  infernals  mountains  clouds  oceans  chemtechs  hextechs  dragons (type unknown)  elders  opp_elders  firstherald  heralds  opp_heralds  firstbaron  barons  opp_barons  firsttower  towers  opp_towers  firstmidtower  firsttothreetowers  turretplates  opp_turretplates  inhibitors  opp_inhibitors  damagetochampions        dpm  damageshare  damagetakenperminute  damagemitigatedperminute  \
0       ESPORTSTMNT01_2690210         complete  None  LCK CL  2022   Spring         0  2022-01-10 07:44:08     1  12.01              1  Blue      top     Soboro  oe:player:38e0af7278d6769d0c81d7c4b47ac1e    Fredit BRION Challengers  oe:team:68911b3329146587617ab2973106e23  Renekton     Karma  Caitlyn  Syndra  Thresh    Lulu        1713       0      2       3        2          9          19          0.0          0.0          0.0         0.0         0.0             0.0               0.0               0.0    0.3152  0.9807          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     0.0         0.0         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         0.0             0.0            15768.0   552.2942     0.278784             1072.3993                  777.7933   
1       ESPORTSTMNT01_2690210         complete  None  LCK CL  2022   Spring         0  2022-01-10 07:44:08     1  12.01              2  Blue      jng     Raptor  oe:player:637ed20b1e41be1c51bd1a4cb211357    Fredit BRION Challengers  oe:team:68911b3329146587617ab2973106e23  Xin Zhao     Karma  Caitlyn  Syndra  Thresh    Lulu        1713       0      2       5        6          9          19          0.0          0.0          0.0         0.0         1.0             0.0               1.0               0.0    0.3152  0.9807          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     0.0         0.0         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         0.0             1.0            11765.0   412.0841     0.208009              944.2732                  650.1576   
2       ESPORTSTMNT01_2690210         complete  None  LCK CL  2022   Spring         0  2022-01-10 07:44:08     1  12.01              3  Blue      mid     Feisty  oe:player:d1ae0e2f9f3ac1e0e0cdcb86504ca77    Fredit BRION Challengers  oe:team:68911b3329146587617ab2973106e23   LeBlanc     Karma  Caitlyn  Syndra  Thresh    Lulu        1713       0      2       2        3          9          19          0.0          0.0          0.0         0.0         0.0             0.0               0.0               0.0    0.3152  0.9807          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     0.0         0.0         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         0.0             0.0            14258.0   499.4046     0.252086              581.6462                  227.7758   
3       ESPORTSTMNT01_2690210         complete  None  LCK CL  2

In [18]:
df[df['datacompleteness'] == 'partial']

gameid datacompleteness                                          url league  year   split  playoffs                 date  game  patch  participantid  side position playername                                   playerid            teamname                                   teamid   champion      ban1     ban2     ban3     ban4     ban5  gamelength  result  kills  deaths  assists  teamkills  teamdeaths  doublekills  triplekills  quadrakills  pentakills  firstblood  firstbloodkill  firstbloodassist  firstbloodvictim  team kpm    ckpm  firstdragon  dragons  opp_dragons  elementaldrakes  opp_elementaldrakes  infernals  mountains  clouds  oceans  chemtechs  hextechs  dragons (type unknown)  elders  opp_elders  firstherald  heralds  opp_heralds  firstbaron  barons  opp_barons  firsttower  towers  opp_towers  firstmidtower  firsttothreetowers  turretplates  opp_turretplates  inhibitors  opp_inhibitors  damagetochampions        dpm  damageshare  damagetakenperminute  \
24      8401-8401_game_1          partial  https://lpl.qq.com/es/stats.shtml?bmid=8401    LPL  2022  Spring         0  2022-01-10 09:24:26     1  12.01              1  Blue      top     shanji  oe:player:708bbc09836488d812e6775a07d785f           Oh My God  oe:team:f4c4528c6981e104a11ea7548630c23       Gwen  Renekton  Lee Sin  Caitlyn    Jayce  Camille        1365       1      5       0        4         13           6          NaN          NaN          NaN         NaN         NaN             0.0               NaN               NaN    0.5714  0.8352          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     NaN         NaN         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         NaN             NaN            11188.0   491.7802     0.279107              585.8022   
25      8401-8401_game_1          partial  https://lpl.qq.com/es/stats.shtml?bmid=8401    LPL  2022  Spring         0  2022-01-10 09:24:26     1  12.01              2  Blue      jng        Aki  oe:player:3faae62fb907ad7cbcf2167beb8b2c7           Oh My God  oe:team:f4c4528c6981e104a11ea7548630c23  Jarvan IV  Renekton  Lee Sin  Caitlyn    Jayce  Camille        1365       1      0       1       13         13           6          NaN          NaN          NaN         NaN         NaN             0.0               NaN               NaN    0.5714  0.8352          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     NaN         NaN         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         NaN             NaN             4426.0   194.5495     0.110416              828.8352   
26      8401-8401_game_1          partial  https://lpl.qq.com/es/stats.shtml?bmid=8401    LPL  2022  Spring         0  2022-01-10 09:24:26     1  12.01              3  Blue      mid      Creme  oe:player:6e7908512cb309a43388bb464a312a2           Oh My God  oe:team:f4c4528c6981e104a11ea7548630c23     Syndra  Renekton  Lee Sin  Caitlyn    Jayce  Camille        1365       1      3       0        5         13           6          NaN          NaN          NaN         NaN         NaN             0.0               NaN               NaN    0.5714  0.8352          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     NaN         NaN         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         NaN             NaN            12577.0   552.8352     0.313755              201.8901   
27      8401-8401_game_1          partial  https:/

In [19]:
df[df['datacompleteness'] == 'ignore']

gameid datacompleteness                                                url league  year   split  playoffs                 date  game  patch  participantid  side position  playername                                   playerid    teamname                                   teamid      champion     ban1          ban2      ban3     ban4    ban5  gamelength  result  kills  deaths  assists  teamkills  teamdeaths  doublekills  triplekills  quadrakills  pentakills  firstblood  firstbloodkill  firstbloodassist  firstbloodvictim  team kpm    ckpm  firstdragon  dragons  opp_dragons  elementaldrakes  opp_elementaldrakes  infernals  mountains  clouds  oceans  chemtechs  hextechs  dragons (type unknown)  elders  opp_elders  firstherald  heralds  opp_heralds  firstbaron  barons  opp_barons  firsttower  towers  opp_towers  firstmidtower  firsttothreetowers  turretplates  opp_turretplates  inhibitors  opp_inhibitors  damagetochampions        dpm  damageshare  \
18300  ESPORTSTMNT06_2210192           ignore  http://matchhistory.na.leagueoflegends.com/en/...    LEC  2022  Spring         0  2022-02-11 20:50:06     1  12.03              1  Blue      top       Armut  oe:player:2d2b174e23f40f5ea3a0c4151349ca3   MAD Lions  oe:team:47ae4d96b0991cd0bc4271a5d03ebbf        Graves    Jayce  Twisted Fate    Akshan     Lulu    Ryze        1620       0      0       2        2          4          14          0.0          0.0          0.0         0.0         1.0             0.0               1.0               0.0    0.1481  0.6667          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     NaN         NaN         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         0.0             0.0            10538.0   390.2963     0.311305   
18301  ESPORTSTMNT06_2210192           ignore  http://matchhistory.na.leagueoflegends.com/en/...    LEC  2022  Spring         0  2022-02-11 20:50:06     1  12.03              2  Blue      jng      Elyoya  oe:player:8a8f18e4b7d0f30c19c5fb99a526951   MAD Lions  oe:team:47ae4d96b0991cd0bc4271a5d03ebbf         Diana    Jayce  Twisted Fate    Akshan     Lulu    Ryze        1620       0      3       3        1          4          14          0.0          0.0          0.0         0.0         1.0             1.0               0.0               0.0    0.1481  0.6667          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     NaN         NaN         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         0.0             1.0             9137.0   338.4074     0.269918   
18302  ESPORTSTMNT06_2210192           ignore  http://matchhistory.na.leagueoflegends.com/en/...    LEC  2022  Spring         0  2022-02-11 20:50:06     1  12.03              3  Blue      mid      Reeker  oe:player:a8ab1ca5ba5183bdad08a3f2eeeb95d   MAD Lions  oe:team:47ae4d96b0991cd0bc4271a5d03ebbf        Irelia    Jayce  Twisted Fate    Akshan     Lulu    Ryze        1620       0      0       4        0          4          14          0.0          0.0          0.0         0.0         0.0             0.0               0.0               0.0    0.1481  0.6667          NaN      NaN          NaN              NaN                  NaN        NaN        NaN     NaN     NaN        NaN       NaN                     NaN     NaN         NaN          NaN      NaN          NaN         NaN     NaN         NaN         NaN     NaN         NaN            NaN                 NaN           NaN               NaN         0.0             0.0             5824.0   215.7037     0.172048   
18303  ESPORTSTMNT06_2210192           ignore  http://matchhistory.na.leagueoflegends.com/en/...    LE

If datacompleteness does not say complete, it means some of the data might be missing, so some of the stats may have no values.

# URL

The column labeled "URL" was intended to link to websites containing information about the matches. However, it is noteworthy that the majority of matches possessing the datacompleteness status of 'complete' lack a corresponding URL.

In [20]:
num_missing = df["url"].isnull().sum()
num_urls = df["url"].notnull().sum()

print("Number of missing URLs: ", num_missing)
print("Number of URLs: ", num_urls)

Number of missing URLs:  126984
Number of URLs:  22416


In [27]:
# create a DataFrame with the URL data
data_url = pd.DataFrame({
    "labels": ["Missing URLs", "Non-Missing URLs"],
    "values": [num_missing, num_urls]
})

# create the pie chart using Plotly Express with dark theme
fig = px.pie(data_url, values='values', names='labels', title='URL Availability',
             template="plotly_dark", color_discrete_sequence=px.colors.sequential.RdBu)

# show the chart
fig.show()

ValueError: Mime type rendering requires nbformat>=4.2.0 but it is not installed

As the dataset reveals an 85.04% absence of URLs in this particular column, it may be imprudent to employ it in subsequent analyses.

# League

The column labeled "League" specifies the origin of the teams participating in the matches.

In [ ]:
df['league'].unique()

array(['LCK CL', 'LPL', 'NLC', 'SL', 'Proving Grounds Circuit', 'UL',
       'PRM', 'LCK', 'LFL', 'LEC', 'LCS', 'LFL2', 'GLL', 'HM', 'ESLOL',
       'EBL', 'LPLOL', 'PGN', 'LCSA', 'LVP DDH', 'TAL', 'TCL', 'CBLOL',
       'LCO', 'LHE', 'GL', 'EL', 'CBLOLA', 'LMF', 'VL', 'LLA', 'HC',
       'LDL', 'LJL', 'PCS', 'VCS', 'UPL', 'LCL', 'NEXO', 'EM', 'LAS',
       'MSI', 'LJLA', 'CT', 'WCS', 'CDF', 'IC', 'DC'], dtype=object)

In total, there are fifty distinct leagues present. Let us determine the leagues that exhibit a greater volume of matches.

In [ ]:
# retrieve the initial ID within each group of duplicated IDs.
df_unique_id = df.groupby('gameid').first().reset_index()

# Group the dataset by the "league" column, calculate the total number of game IDs for each league, and arrange the result in descending order.
df_unique_id = df_unique_id.groupby('league').agg({'gameid':'count'}).sort_values(by = 'gameid',
                                                                                  ascending = False).reset_index()

# Generate a dataframe of the top 10 leagues based on the highest number of matches played.
df_top_10 = df_unique_id.head(10)

In [ ]:
df_top_10

league  gameid
0                      LDL     942
1                      LPL     786
2  Proving Grounds Circuit     562
3                     LCSA     540
4                      LCK     467
5                      UPL     412
6                       SL     410
7                   LCK CL     394
8                      NLC     383
9                      PRM     367

In [ ]:
fig = px.bar(df_top_10.sort_values(by = 'gameid', ascending = True), y = 'league', x = 'gameid',
             template = 'plotly_dark', orientation = 'h',
             text_auto = '.3s', color_discrete_sequence=px.colors.sequential.RdBu)

fig.update_layout(
    title = {
        'text': "Top 10 Leagues: Most Matches Played"},
    xaxis_title = 'Number of Matches',
    yaxis_title = 'Leagues',
    width = 1200,
    height = 700,
    font_family="Helvetica",
    font_color="White",
    font = dict(size = 18),
    title_font_family="Helvetica",
    title_font_color= "White",
    bargap = 0.4,
    uniformtext_minsize = 8)


fig.show()

# Year

In [ ]:
df['year'].unique()

array([2022, 2023], dtype=int64)

In [ ]:
registries_in_2022 = df[df['year'] == 2022].shape[0]
registries_in_2023 = df[df['year'] == 2023].shape[0]

print("Registries from 2022: ", registries_in_2022)
print("Registries from 2023: ", registries_in_2023)

Registries from 2022:  145332
Registries from 2023:  4068


This dataset comprises data for the years 2022 and 2023, with the larger portion of the data corresponding to the year 2022.

# Split

In [ ]:
df['split'].unique()

array(['Spring', None, 'Split 1', 'Winter', 'Opening', 'Summer',
       'Champ 1', 'Split 2', 'Closing', '2022', 'Pro-Am', 'Fall',
       'Champ 2'], dtype=object)

In [ ]:
df_split = df['split'].value_counts().reset_index()
df_split.rename(columns = {'index':  'Split',
                           'split':'matches'}, inplace = True)

In [ ]:
df_split['cumulative_matches'] = df_split['matches'].cumsum()
df_split

Split  matches  cumulative_matches
0    Summer    45216               45216
1    Spring    42096               87312
2   Split 1     7224               94536
3   Split 2     5064               99600
4    Winter     2448              102048
5   Champ 2     1524              103572
6      Fall     1392              104964
7   Champ 1     1200              106164
8   Opening     1080              107244
9   Closing     1080              108324
10   Pro-Am      864              109188
11     2022      240              109428

In [ ]:
fig = px.bar(df_split.sort_values(by = 'matches', ascending = True),
             y = 'Split', x = 'matches',
             template = 'plotly_dark', orientation = 'h',
             text_auto = '.3s', color_discrete_sequence=px.colors.sequential.RdBu)

fig.update_layout(
    title = {
        'text': "Leagues with most games played"},
    xaxis_title = 'Number of Matches',
    yaxis_title = 'Splits',
    width = 1200,
    height = 700,
    font_family="Helvetica",
    font_color="White",
    font = dict(size = 18),
    title_font_family="Helvetica",
    title_font_color= "White",
    bargap = 0.4,
    uniformtext_minsize = 8)


fig.show()

The most played matches are on Summer and Spring and there are 240 matches played with the split flag called 2022.

# Playoffs

In [ ]:
df['playoffs'].value_counts()

playoffs
0    123828
1     25572
Name: count, dtype: int64

It is clear to see how there are much more registries of non play-offs games, what was expected considering the time they're played.

# Date

The first game started at 2022-01-10 and the last one in 2022-12-27.

In [ ]:
df['date'].head(1)

0    2022-01-10 07:44:08
Name: date, dtype: object

In [ ]:
df['date'].tail(1)

149399    2022-12-27 12:43:43
Name: date, dtype: object

# Game

In [ ]:
df['game'].value_counts(ascending = False)

game
1    102024
2     26736
3     13908
4      4644
5      2088
Name: count, dtype: int64

It is possible to see that there are less amount of games registries with higer name of games.

# Patch

In [ ]:
df['patch'].nunique()

22

In this dataset we have 22 patches, therefore, the game was updated 22 times during competitive matches in League of Legends.

In [ ]:
df.groupby(by = 'patch').agg({'patch':'count'})

patch
patch       
12.01   9624
12.02  11616
12.03  10524
12.04  15888
12.05  14940
12.06   2244
12.07    984
12.08   2172
12.09   4224
12.10   8088
12.11  11328
12.12  16212
12.13  11928
12.14   7128
12.15   5520
12.16   3420
12.17    288
12.18   6336
12.19   2400
12.20   2196
12.21   1200
12.23   1032

In [ ]:
fig = px.bar(df.groupby(by = 'patch').agg({'patch':'count'}),
             y = 'patch', x = 'patch',
             template = 'plotly_dark', orientation = 'h',
             text_auto = '.3s', color_discrete_sequence=px.colors.sequential.RdBu)

fig.update_layout(
    title = {
        'text': "Leagues with most games played"},
    xaxis_title = 'Number of Matches',
    yaxis_title = 'Splits',
    width = 1200,
    height = 700,
    font_family="Helvetica",
    font_color="White",
    font = dict(size = 18),
    title_font_family="Helvetica",
    title_font_color= "White",
    bargap = 0.4,
    uniformtext_minsize = 8)


fig.show()

ValueError: Mime type rendering requires nbformat>=4.2.0 but it is not installed